apps/gitness/src/pages/pipeline/project-pipeline-list.tsx

```diff
@@ -15,11 +15,11 @@ import { PageResponseHeader } from '../../types'
 import { getExecutionStatus } from '../../utils/execution-utils'
 import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
 import { timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
-import { useDebouncedQuery } from '../../hooks/useQuery'
+import { useDebouncedQueryState } from '../../hooks/useQuery'
 
 export default function ProjectPipelinesPage() {
   const spaceId = useGetSpaceURLParam()
-  const [query, setQuery] = useDebouncedQuery()
+  const [query, setQuery] = useDebouncedQueryState({ queryKey: 'query' })
```

Comment:
let's follow the same API as `useQueryState`. 
We should be able to call this hook like `useDebouncedQueryState('key', 'value')` and `useDebouncedQueryState('key', { defaultValue: 'value' })`

Reply:
Done